File: day2/ques8.cpp
#include <iostream>
using namespace std;

int main() {
    int num, reversed = 0;

    cout << "Enter a number: ";
    cin >> num;

    int temp = num;  // Save original

    // Reverse the number
    while (temp != 0) {
        int lastDigit = temp % 10;
        reversed = reversed * 10 + lastDigit;
        temp = temp / 10;
    }

    // Check palindrome
    if (num == reversed) {
        cout << num << " is a Palindrome!" << endl;
    } else {
        cout << num << " is NOT a Palindrome!" << endl;
    }

    return 0;
}

File: day2 python/ques8.py
# Take input from the user
num = int(input("Enter a number: "))

# Store the original number in a temporary variable
temp = num
reverse_num = 0

# Loop to reverse the number
while temp > 0:
    digit = temp % 10
    reverse_num = (reverse_num * 10) + digit
    temp //= 10

# Compare original number with the reversed number
if num == reverse_num:
    print(f"{num} is a palindrome number.")
else:
    print(f"{num} is not a palindrome number.")
